zipline/protocol.py

```diff
@@ -225,14 +227,17 @@ def __repr__(self):
 
 
 class Position(object):
-
-    def __init__(self, sid):
-        self.sid = sid
+    @expect_types(asset=Asset)
+    def __init__(self, asset):
+        self.asset = asset
         self.amount = 0
         self.cost_basis = 0.0  # per share
         self.last_sale_price = 0.0
         self.last_sale_date = None
 
+        # for backwards compatibility
+        self.sid = asset
```

Comment:
I'd probably prefer to use a property here unless you're concerned about performance overhead.  As this stands, it's possible for `sid` and `asset` to diverge if someone mutates one and not the other after construction.  (Admittedly, doing that is probably a bad idea.)